ra-optimisation/linear-rao/src/main/java/com/farao_community/farao/linear_rao/SystematicAnalysisEngine.java

```diff
@@ -166,13 +176,20 @@ private void checkSensiResults(LinearRaoData linearRaoData, SystematicSensitivit
      */
     private void setResults(LinearRaoData linearRaoData, SystematicSensitivityAnalysisResult systematicSensitivityAnalysisResult, Map<String, Double> loopflows) {
         linearRaoData.setSystematicSensitivityAnalysisResult(systematicSensitivityAnalysisResult);
+        //set functional cost
         double minMargin = getMinMargin(linearRaoData, systematicSensitivityAnalysisResult);
         linearRaoData.getCracResult().setFunctionalCost(-minMargin);
-        linearRaoData.getCracResult().setVirtualCost(fallbackMode ? linearRaoParameters.getFallbackOvercost() : 0);
-        if (isLoopflowViolation()) {
-//            linearRaoData.getCracResult().setCost(Double.POSITIVE_INFINITY); //todo: set a high cost if loopflow constraint violation => use "virtual cost" in the future
-        }
+
+        //update virtual cost
+        double virtualCost = linearRaoData.getCracResult().getVirtualCost();
+        double fallbackModeVirtualCost = fallbackMode ? linearRaoParameters.getFallbackOvercost() : 0.0;
+        linearRaoData.getCracResult().setVirtualCost(virtualCost + fallbackModeVirtualCost);
+
         linearRaoData.getCracResult().setNetworkSecurityStatus(minMargin < 0 ? CracResult.NetworkSecurityStatus.UNSECURED : CracResult.NetworkSecurityStatus.SECURED);
+        if (linearRaoParameters.getExtendable().getLoopflowViolationCost() == 0.0 && isLoopflowViolation()) {
```

Comment:
Same, why do you test that 'linearRaoParameters.getExtendable().getLoopflowViolationCost() == 0.0' ?

Reply:
The “cost=0” routine is different from the not zero one in linear optimizer.